lib/elixir/lib/enum.ex

```diff
@@ -3050,6 +3068,12 @@ defmodule Enum do
     |> div(2)
   end
 
+  # TODO: Remove me on v2.0
+  def sum(%{__struct__: Range, first: first, last: last} = range) do
```

Comment:
I believe we don't need this clause because it will fallback to the protocol.

Reply:
You're right
```
iex(4)> Enum.sum(old_range)    
55
```